### PR TITLE
CPD-10113: Updated to swift 4.0

### PR DESCRIPTION
**Pull request by AlkiviadisPapadakis-je**
https://jira.just-eat.net/browse/CPD-10113

### CODE REVIEW

File: Example/Podfile.lock [hunk 1]
```diff
@@ -1,51 +1,63 @@
 PODS:
-  - Firebase/Core (3.14.0):
-    - FirebaseAnalytics (= 3.7.0)
-    - FirebaseCore (= 3.5.1)
-  - Firebase/RemoteConfig (3.14.0):
+  - Firebase/Core (4.7.0):
```
Comment (federicocappelli-je) [inline Example/Podfile.lock]: why this huge version bump?

File: README.md [hunk 2]
```diff
@@ -98,6 +98,8 @@ it, simply add the following line to your Podfile:
 ```ruby
 pod "JustTweak"
 ```
+## Requirements
+JustTweak supports swift 4.0 and xCode 9
```
Comment (federicocappelli-je) [inline README.md]: Xcode ;)

File: JustTweak/Classes/Core/EphemeralConfiguration.swift [hunk 3]
```diff
@@ -5,6 +5,7 @@
 
 import Foundation
 
+@objcMembers
 @objc(JEEphemeralConfiguration) public class EphemeralConfiguration: NSObject, MutableTweaksConfiguration {
```
Comment (markcousins-je) [inline JustTweak/Classes/Core/EphemeralConfiguration.swift]: Do we need @objc if using @objcmembers?

File: Example/Podfile.lock (hunk 1 above)
Comment (AlkiviadisPapadakis-je) [reply Example/Podfile.lock]: because pod update.

Comment (albertodebortoli-je) [reply Example/Podfile.lock]: It's ok, it's just for the demo app. It's not a dependency in the podspec.

File: JustTweak/Classes/Core/EphemeralConfiguration.swift (hunk 3 above)
Comment (albertodebortoli-je) [reply JustTweak/Classes/Core/EphemeralConfiguration.swift]: I would remove all the things JE-prefixed, as done for other repositories. Let's live in the future.

Comment (AlkiviadisPapadakis-je) [reply JustTweak/Classes/Core/EphemeralConfiguration.swift]: I add both so I don't break the client using it. If we decide to remove, happy to do.

File: Example/Podfile.lock (hunk 1 above)
Comment (federicocappelli-je) [reply Example/Podfile.lock]: LoL, so it is ok test the app with completely different lib version? this is a HUGE RED warning for me.

File: JustTweak/Classes/Core/EphemeralConfiguration.swift (hunk 3 above)
Comment (federicocappelli-je) [reply JustTweak/Classes/Core/EphemeralConfiguration.swift]: I would propose to not change the interface, leave it

**Pull request merged**